internal/hammer/hammer.go

```diff
@@ -71,23 +73,30 @@ var (
 	httpTimeout = flag.Duration("http_timeout", 30*time.Second, "Timeout for HTTP requests")
 	forceHTTP2  = flag.Bool("force_http2", false, "Use HTTP/2 connections *only*")
 
-	hc = &http.Client{
-		Transport: &http.Transport{
-			MaxIdleConns:        256,
-			MaxIdleConnsPerHost: 256,
-			DisableKeepAlives:   false,
-		},
-		Timeout: *httpTimeout,
-	}
+	hc *http.Client
 )
 
 func main() {
 	klog.InitFlags(nil)
 	flag.Parse()
 
+	hc = &http.Client{
+		Transport: &http.Transport{
+			MaxIdleConns:        *numWriters,
+			MaxIdleConnsPerHost: *numWriters,
```

Comment:
Is it necessary to have both? I don't think it's possible to run into MaxIdleConnsPerHost without also running into MaxIdleConns?

Reply:
It's necessary unfortunately - the default value for `MaxIdleConnsPerHost` is `2`: https://go.dev/src/net/http/transport.go#L58